File: Models/OperationResult.cs
using financial_manager.Models.Enums;

namespace financial_manager.Models
{
    public class OperationResult
    {
        public bool Success { get; set; }
        public HttpResponseCode HttpResponseCode { get; set; }
        public string? Message { get; set; }

        public OperationResult(bool success, string? message = null)
        public OperationResult(bool isSuccess, HttpResponseCode httpResponseCode, string? message = null)
        {
            Success = success;
            HttpResponseCode = httpResponseCode;
            Message = message;
        }
    }
}

File: UI/financial_manager/src/app/models/operation-result.ts
import { Category } from './category';
import { HttpResponseCode } from './enums/http-response-code';
import { Transaction } from './transaction';
import { User } from './user';

export class OperationResult<T = User | Category | Transaction> {
  isSuccess: boolean;
  httpResponseCode: HttpResponseCode;
  message?: string;
  data?: T[];

  constructor(
    isSuccess: boolean,
    httpResponseCode: HttpResponseCode,
    message?: string,
    data?: T[]
  ) {
    this.isSuccess = isSuccess;
    this.httpResponseCode = httpResponseCode;
    this.message = message;
    this.data = data;
  }
}
